lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_state_machine.rb

```diff
@@ -83,7 +83,7 @@ def process_state(f, message, args)
         elsif @workspace.root['ae_result'] == 'skip'
           $miq_ae_logger.warn("Skipping State=[#{f['name']}]")
           return set_next_state(f, message)
-        elsif %w(retry restart).include?(@workspace.root['ae_result'])
+        elsif %w(retry restart async_launch).include?(@workspace.root['ae_result'])
```

Comment:
Is there an issue with incrementing the retry count on retry and async_launch?  What happens if the playbook requests a retry on completion?  Do we need to reset the retry count at some point?

Reply:
@gmcculloug Its needed to make sure that the playbook method ends and doesn't linger forever.